### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 1]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
...
 
     public:
+      /**
+       * @brief Constructs a Key
+       *
+       */
       Key() : m_uniqueAddress(this) {}
 
+      /**
+       * @brief Compare with \p other #Azure::Core::Context::Key for equality.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:61]: 1. Compares ...
2. Remove self-reference (`#Azure::Core::Context::Key`)

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 2]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
...
+       * @param other Other #Azure::Core::Context::Key to compare with.
+       * @return `true` if #Azure::Core::Context::Key instances are equal; otherwise, `false`.
+       */
       bool operator==(Key const& other) const
       {
         return this->m_uniqueAddress == other.m_uniqueAddress;
       }
 
+      /**
+       * @brief Compare with \p other #Azure::Core::Context::Key for equality.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:71]: 1. Compares...
2. Remove self-reference

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 3]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
...
+       * @param other Other #Azure::Core::Context::Key to compare with.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:72]: self-reference

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 4]
```diff
@@ -204,12 +219,12 @@ namespace Azure { namespace Core {
...
+     * @return `true` if this context is cancelled; otherwise, `false`.
      */
     bool IsCancelled() const noexcept { return GetDeadline() < std::chrono::system_clock::now(); }
 
     /**
-     * @brief Throws an exception if the context is cancelled.
+     * @brief Throws an #Azure::Core::OperationCancelledException if the context is cancelled.
      *
      */
     void ThrowIfCancelled() const
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:230]: Here an in other places that throw: use `@throw` doxygen tag.

File: sdk/core/azure-core/inc/azure/core/operation.hpp [hunk 5]
```diff
@@ -72,25 +72,29 @@ namespace Azure { namespace Core {
...
+     * @brief Destructs the Operation.
+     *
+     */
     virtual ~Operation() {}
 
     /**
      * @brief Final result of the long-running operation.
      *
-     * @return Response<T> the final result of the long-running operation.
+     * @return T the final result of the long-running operation.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation.hpp:84]:
```suggestion
     * @return The final result of the long-running operation.
```

File: sdk/core/azure-core/inc/azure/core/exception.hpp [hunk 6]
```diff
@@ -66,10 +66,10 @@ namespace Azure { namespace Core {
     std::unique_ptr<Azure::Core::Http::RawResponse> RawResponse;
 
     /**
-     * @brief Construct a new Request Failed Exception object.
+     * @brief Constructs a new RequestFailedException object.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/exception.hpp:69]:
```suggestion
     * @brief Constructs a new `RequestFailedException` object.
```

File: sdk/core/azure-core/inc/azure/core/context.hpp (hunk 2 above)
Comment (antkmsft) [reply sdk/core/azure-core/inc/azure/core/context.hpp:71]: If you want to use the name of the class, such as "Compare with another `Key`", backtick the `Key`.

File: sdk/core/azure-core/inc/azure/core/match_conditions.hpp [hunk 7]
```diff
@@ -20,6 +20,10 @@ namespace Azure {
  */
 struct MatchConditions
 {
+  /**
+   * @brief Destructs the MatchConditions.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/match_conditions.hpp:24]:
```suggestion
   * @brief Destructs the `MatchConditions`.
```

Please, review other places her yourself and backtick all the places where we don't want to self-refrence, but need to use the name of self class.

File: sdk/core/azure-core/inc/azure/core/url.hpp [hunk 8]
```diff
@@ -166,7 +166,7 @@ namespace Azure { namespace Core {
      * will be encoded before adding to the URL. Use \p isValueEncoded = true when the
      * value is already encoded.
      *
-     * @remark This function overrides the value of existing query parameters.
+     * @remark Overrides the value of existing query parameters.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/url.hpp:169]: `@note`?

**Review by antkmsft — CHANGES_REQUESTED** [review]
Please ally my feedback.

BTW, when you will be updating `DateTime` (and if not in this PR, it is fine), can you please mention that it is `std::chrono::time_point` and link https://en.cppreference.com/w/cpp/chrono/time_point ?

**Comment by RickWinter** [comment]
> Please ally my feedback.
> 
> BTW, when you will be updating `DateTime` (and if not in this PR, it is fine), can you please mention that it is `std::chrono::time_point` and link https://en.cppreference.com/w/cpp/chrono/time_point ?

DateTime was not in my assigned namespace.  If it hasn't been done by the owner I can look to help out later tonight

**Comment by ahsonkhan** [comment]
Part of https://github.com/Azure/azure-sdk-for-cpp/issues/2260

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 9]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
...
     public:
+      /**
+       * @brief Constructs a Key
+       *
+       */
       Key() : m_uniqueAddress(this) {}
 
+      /**
+       * @brief Compare with \p other `Azure::Core::Context::Key` for equality.
+       * @param other Other `Azure::Core::Context::Key` to compare with.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:62]:
```suggestion
       * @brief Compares with \p other `Key` for equality.
       * @param other Other `Key` to compare with.
```

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 10]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
...
+       * @return `true` if instances are equal; otherwise, `false`.
+       */
       bool operator==(Key const& other) const
       {
         return this->m_uniqueAddress == other.m_uniqueAddress;
       }
 
+      /**
+       * @brief Compare with \p other `Azure::Core::Context::Key` for equality.
+       * @param other Other `Azure::Core::Context::Key` to compare with.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:72]:
```suggestion
       * @brief Compares with \p other `Key` for inequality.
       * @param other Other `Key` to compare with.
```

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 11]
```diff
@@ -154,27 +168,28 @@ namespace Azure { namespace Core {
...
 
     /**
-     * @brief Get a deadline associated with this context or the branch of contexts this context
+     * @brief Gets the deadline for this context or the branch of contexts this context
      * belongs to.
      *
-     * @return A deadline associated with the context found; `Azure::DateTime::max()` value if a
-     * specific value can't be found.
+     * @return The deadline associated with the context;
+     * @return `Azure::DateTime::max()` if no deadline is specified.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:175]: it is `@retval` that would handle multiple return values. `@return` won't.
I suggest we use `@return` here as before.
With `@retval` I am not sure what to put for the value before describing it.

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 12]
```diff
@@ -154,27 +168,28 @@ namespace Azure { namespace Core {
...
      * branch of contexts this context belongs to.
      *
      * @tparam T The type of the value to be retrieved.
      * @param key A key associated with a context to find.
      * @param outputValue A reference to the value corresponding to the \p key to be set, if found
      * within the context tree.
      *
-     * @return If found, returns `true`, with \p outputValue set to the value associated with the
-     * key found; otherwise returns `false`.
+     * @return `true` if Key is found, with \p outputValue set to the value associated with the
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:189]:
```suggestion
     * @return `true` if \p key is found, with \p outputValue set to the value associated with the
```

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 13]
```diff
@@ -22,13 +22,13 @@
...
   /**
-   * @brief An exception that gets thrown when some operation is cancelled.
+   * @brief An exception thrown when an operation is cancelled.
    *
    */
   class OperationCancelledException final : public std::runtime_error {
   public:
     /**
-     * @brief Construct with message string as description.
+     * @brief Constructs an OperationCancelledException with message string as the description.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:31]:
```suggestion
     * @brief Constructs an `OperationCancelledException` with message string as the description.
```

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 14]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
...
-     * @brief A context key.
+     * @brief A key used to store and retrieve data in a Context object.
      *
      */
     class Key final {
       Key const* m_uniqueAddress;
 
     public:
+      /**
+       * @brief Constructs a Key
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:55]:
```suggestion
       * @brief Constructs a `Key`.
```

File: sdk/core/azure-core/inc/azure/core/context.hpp [hunk 15]
```diff
@@ -44,20 +44,34 @@ namespace Azure { namespace Core {
   class Context final {
   public:
     /**
-     * @brief A context key.
+     * @brief A key used to store and retrieve data in a Context object.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/context.hpp:47]:
```suggestion
     * @brief A key used to store and retrieve data in an #Azure::Core::Context object.
```

File: sdk/core/azure-core/inc/azure/core/operation.hpp [hunk 16]
```diff
@@ -72,25 +72,29 @@ namespace Azure { namespace Core {
     }
 
   public:
+    /**
+     * @brief Destructs the Operation.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation.hpp:76]:
```suggestion
     * @brief Destructs the `Operation`.
```

File: sdk/core/azure-core/inc/azure/core/operation.hpp [hunk 17]
```diff
@@ -122,15 +126,16 @@ namespace Azure { namespace Core {
     }
 
     /**
-     * @brief Returns true if the long-running operation completed successfully and has produced a
-     * final result.  The final result is accessible from Value().
+     * @brief Checks if the long-running operation completed successfully and has produced a
+     * final result.
+     * @note The final result is accessible from Value().
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation.hpp:131]:
```suggestion
     * @note The final result is accessible from `Value()`.
```

File: sdk/core/azure-core/inc/azure/core/operation.hpp [hunk 18]
```diff
@@ -142,9 +147,9 @@ namespace Azure { namespace Core {
     }
 
     /**
-     * @brief Calls the server to get updated status of the long-running operation.
+     * @brief Gets updated status of the long-running operation.
      *
-     * @param context #Azure::Core::Context allows the user to cancel the long-running operation.
+     * @param context #Azure::Core::Context allows canceling of the long-running operation.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation.hpp:152]:
```suggestion
     * @param context #Azure::Core::Context allows cancelling of the long-running operation.
```
Super nit, just for uniformity; We use LL in other places.

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 19]
```diff
@@ -27,30 +27,30 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * @brief Construct an #Azure::Core::OperationStatus with \p value.
+     * @brief Constructs an #Azure::Core::OperationStatus with \p value.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:30]:
```suggestion
     * @brief Constructs an `OperationStatus` with \p value.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 20]
```diff
@@ -27,30 +27,30 @@ namespace Azure { namespace Core {
...
      *
      * @param value The value to initialize with.
      */
     explicit OperationStatus(const std::string& value) : m_value(value) {}
     /**
-     * @brief Construct an #Azure::Core::OperationStatus with \p value.
+     * @brief Constructs an #Azure::Core::OperationStatus with \p value.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:36]:
```suggestion
     * @brief Constructs an `OperationStatus` with \p value.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 21]
```diff
@@ -27,30 +27,30 @@ namespace Azure { namespace Core {
...
      *
      * @param value The value to initialize with.
      */
     explicit OperationStatus(std::string&& value) : m_value(std::move(value)) {}
     /**
-     * @brief Construct an #Azure::Core::OperationStatus with \p value.
+     * @brief Constructs an #Azure::Core::OperationStatus with \p value.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:42]:
```suggestion
     * @brief Constructs an `OperationStatus` with \p value.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 22]
```diff
@@ -27,30 +27,30 @@ namespace Azure { namespace Core {
...
      *
      * @param value The value to initialize with.
      */
     explicit OperationStatus(const char* value) : m_value(value) {}
 
     /**
-     * @brief Compare two #Azure::Core::OperationStatus objects for equality.
+     * @brief Compares two #Azure::Core::OperationStatus objects for equality.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:49]:
```suggestion
     * @brief Compares two `OperationStatus` objects for equality.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 23]
```diff
@@ -27,30 +27,30 @@ namespace Azure { namespace Core {
...
      *
      * @param other An #Azure::Core::OperationStatus to compare with.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:51]:
```suggestion
     * @param other An `OperationStatus` to compare with.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp (hunk 22 above)
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:49]:
```suggestion
     * @brief Compares two `OperationStatus` objects for equality.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 24]
```diff
@@ -59,16 +59,16 @@ namespace Azure { namespace Core {
     }
 
     /**
-     * @brief Compare two #Azure::Core::OperationStatus objects for equality.
+     * @brief Compares two #Azure::Core::OperationStatus objects for equality.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:62]:
```suggestion
     * @brief Compares two `OperationStatus` objects for equality.
```

File: sdk/core/azure-core/inc/azure/core/operation_status.hpp [hunk 25]
```diff
@@ -59,16 +59,16 @@ namespace Azure { namespace Core {
...
      *
      * @param other A #Azure::Core::OperationStatus to compare with.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/operation_status.hpp:64]:
```suggestion
     * @param other An `OperationStatus` to compare with.
```

File: sdk/core/azure-core/inc/azure/core/paged_response.hpp [hunk 26]
```diff
@@ -41,19 +41,19 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * @brief Destructor.
+     * @brief Destructs PagedResponse.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/paged_response.hpp:44]:
```suggestion
     * @brief Destructs `PagedResponse`.
```

File: sdk/core/azure-core/inc/azure/core/paged_response.hpp [hunk 27]
```diff
@@ -70,17 +70,19 @@ namespace Azure { namespace Core {
...
      *
      */
     bool HasPage() const { return m_hasPage; }
 
     /**
-     * @brief Move to the next page of the response.
+     * @brief Moves to the next page of the response.
      *
-     * @remark Calling this method on the last page will set #HasPage() to false.
+     * @note Calling this method on the last page will set #HasPage() to false.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/paged_response.hpp:83]:
```suggestion
     * @note Calling this method on the last page will set `HasPage()` to `false`.
```

File: sdk/core/azure-core/inc/azure/core/url.hpp [hunk 28]
```diff
@@ -3,7 +3,7 @@
 
 /**
  * @file
- * @brief Define an Uniform Resource Location (URL).
+ * @brief Uniform Resource Location (URL).
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/url.hpp:6]:
```suggestion
 * @brief Uniform Resource Locator (URL).
```

File: sdk/core/azure-core/inc/azure/core/uuid.hpp [hunk 29]
```diff
@@ -42,8 +42,8 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * Gets UUID as a string.
-     * @details A string is in canonical format (4-2-2-2-6 lowercase hex and dashes only)
+     * @brief Gets Uuid as a string.
```
Comment (antkmsft) [inline sdk/core/azure-core/inc/azure/core/uuid.hpp:45]:
```suggestion
     * @brief Gets `Uuid` as a string.
```

**Review by antkmsft — APPROVED** [review]

**Pull request merged**